float64 distance
float64[2] start
float64[2] end
float64 angle
